Button Component, Button Styling & Functionality

Hi-Van opened the pull request:
**"components"** folder -> hosts all components created within (Change-Van) branch

**"Button.jsx"** & **"button.css"** -> Added _<Button>_ command, separate button profiles and sizes can be created, removed, or edited inside **"button.jsx"**. **"button.css"** hosts all styling for each button profile created within **"button.jsx"**.

**App.js** -> Added two sample buttons for testing labeled: _download button_ & _preview button_. Buttons can be found by comments above them. Both buttons utilize HTML commands to call downloads and URL pages, which is subject to change in the future.

Hi-Van committed:
Button Component

Customizable Button component, still needs a download() and preview() function

Hi-Van committed:
CSS Changes and Button Functionality

button.css -> Changed button styling from rectanges to circles

App.js -> Commented on Download & Preview Button code, and spaced them from the rest of the html. Changed buttons from text to .PNG icons

public -> Added download_icon.png & preview_icon.png

kipply commented on src/components/button.css:
```diff
@@ -0,0 +1,47 @@
+.btn--small {
+    padding: 8px 8px;
+    font-size: 14px;
+}
```
trailing new line

kipply commented on src/App.js:
```diff
@@ -20,6 +21,24 @@ function App() {
+
+        {/*download button, redownloads download icon on click*/}
+        <Button>
+        <a href='download_icon.png' download>
```
to better componentize, pass the link and icon as props https://reactjs.org/docs/components-and-props.html

kipply replied:
the link needs to wrap the image for the image to actually be a link!

kipply commented on src/components/button.css:
```diff
@@ -0,0 +1,47 @@
+}
+
+/*Styling for buttons from BTNSTYLE array in Button.jsx file*/
+.btn--download--solid{
```
why double dashes in the naming? one is fairly sufficient

ZenaMel replied:
This is BEM notation...

kipply commented on src/components/button.css:
```diff
@@ -0,0 +1,47 @@
+:root {
+    /*Colors can be asigned per variable name here*/
+    --white: #ffffff;
```
no need to var `white`, it can be used as `color: white`

kipply commented on src/components/button.css:
```diff
@@ -0,0 +1,47 @@
+:root {
+    /*Colors can be asigned per variable name here*/
+    --white: #ffffff;
+    --download: #37c8ab;
```
use colours that align with our colour scheme! maybe one light button with dark text and one dark button with light text?

kipply commented on src/components/button.css:
```diff
@@ -0,0 +1,47 @@
+
+.btn {
+    /*General Button Format*/
+    font-family: 'Fira Sans', sans-serif;
```
no need, this should inherit

kipply commented on src/components/button.css:
```diff
@@ -0,0 +1,47 @@
+
+/*Sizing for buttons from BTNSIZE array in Button.jsx file*/
+.btn--medium {
+    height: 70px;
```
check this on mobile! Same with padding and font-size. https://stackoverflow.com/questions/9174669/best-practice-font-size-for-mobile

Hi-Van committed:
DownloadButton & PreviewButton

Added a DownloadButton and PreviewButton component

kipply commented on src/App.js:
```diff
@@ -20,6 +21,24 @@ function App() {
+          <a href="https://github.com/the-template-project/root">
+          <img src="preview_icon.png" alt=""/>
+          </a>
+        </Button>
```
great job with the download and preview buttons. Could you use those as examples in `App.js` instead of this?

Hi-Van replied:
I think everything that's in the App div is going to be in the Homepage tag. Therefore, I only kept them as examples in the **button.jsx** . I thought it would be better because the examples could just be copied and pasted from the **button.jsx** file, making them more easily accessible. I saw it as, if I were to put them into the **App.js** file directly, it would not properly format into the homepage, it would still appear on screen and be functional, but would only format around the homepage rather than inside of it.

kipply replied:
Sorry, I meant instead of a `<Button>` elem to use a `<PreviewButton>` elem. It'll save a lot of space in the file and be more relevant as those are the buttons we have a planned used for.

kipply commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,62 @@
+export const DownloadButton = ({file}) => {
+    return  <a href={file} download>
+        <Button>
+        <img src="download_icon.png" alt=""/>
```
good thinking!

ZenaMel replied:
nice, I'll keep that in mind for my own stuff.

Hi-Van committed:
Merge branch 'master' into Change-Van

Hi-Van committed:
added darkmode & lightmode for buttons

Added darkmode & lightmode download and preview button

Added button examples into button.jsx and App.js

Hi-Van committed:
Reset to App.js

Hi-Van committed:
Merge branch 'Change-Van' of https://github.com/the-template-project/root into Change-Van

Hi-Van committed:
Fixed Node and Linting tests

fixed npm run build

fixed import error and trailing extra line

Hi-Van committed:
Social Button

Social button implimentation, documentation in **button.jsx** file

Hi-Van committed:
Stuff with Proptype

Installed Proptype in JSON file

Hi-Van committed:
Added Social Buttons to Homepage

Added social buttons to homepage

Hi-Van committed:
Integrated Font Awesome Icon Library

Integrate Font Awesome 5 Icon Library in Package JSON files

Change Icons for Download & Preview Buttons in button.jsx

Set button width and height to auto in button.css

Remove Local Download and Preview Icons in Public folder

ZenaMel commented on src/Pages/Homepage.js:
```diff
@@ -13,20 +15,8 @@ function Homepage() {
-          >
-            <img src={twitterIcon} alt="fb icon" />
-          </a>
+          <SocialButton link="https://www.facebook.com/" icon={fbIcon}> </SocialButton>
```
looks DRY to me. nice.

ZenaMel commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,101 @@
+  );
+};
+
+/* Defualt download button, is set to defualt sizing. Needs a file passed in to download
```
typo.

ZenaMel replied:
*2x Defualt --> Default

ZenaMel commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,101 @@
+  </a>
+);
+
+/* Defualt preview button, is set to defualt sizing. Needs a page location to direct towards
```
Again. Typo.

ZenaMel commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,101 @@
+);
+
+/* Defualt preview button, is set to defualt sizing. Needs a page location to direct towards
+Use example: <PreviewButton Page={"https://URL-LINK.COM or HTML-PAGE"}></PreviewButton> */
```
I'd either say "Usage: ..." or "Example: ...". Use example is grammatically confusing

ZenaMel commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,101 @@
+  </a>
+);
+
+/* Defualt DARKMODE download button, is set to defualt sizing. Needs a file passed in to download
```
typo

ZenaMel commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,101 @@
+  </a>
+);
+
+/* Defualt DARKMODE preview button, is set to defualt sizing. Needs a page location to direct toward
```
typo

ZenaMel replied:
Kinda every "default" is typoed. Won't add comments to the next typos of this type.

ZenaMel commented on src/components/button.css:
```diff
@@ -0,0 +1,50 @@
+:root {
+    /*Colors can be asigned per variable name here*/
+    --lightmode: #E9ECEF;
```
good

ZenaMel commented on src/components/button.css:
```diff
@@ -0,0 +1,50 @@
+}
+
+.btn {
+    /*General Button Format*/
```
not a helpful comment imo.

ZenaMel commented on src/components/button.css:
```diff
@@ -0,0 +1,50 @@
+}
+
+.btn:hover {
+    /*Button interaction upon hovering*/
```
also... knowledge over pseudoselectors can be assumed.

ZenaMel commented on src/components/button.css:
```diff
@@ -0,0 +1,50 @@
+}
+
+/*Sizing for buttons from BTNSIZE array in Button.jsx file*/
+.btn-defualt {
```
eww... typo.

ZenaMel replied:
Critical typo

ZenaMel commented on src/components/button.css:
```diff
@@ -0,0 +1,50 @@
+.btn-defualt {
+    height: auto;
+    width: auto;
+    padding: 8px 8px;
```
I haven't seen the overall design yet but a "squary" button is fairly uncommon. If this is in fact a floating action button I'd suggest going for a fully rounded design. Mostly a preferential decision though.

ZenaMel commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,101 @@
+
+/* Button Sizing Array
+Sizing can be created or renamed here */
+const BTNSIZE = ['btn-defualt'];
```
Critical typo

Hi-Van committed:
Learned how to spell "Default"
